tests/descriptor_tests/test_downloadables.py

```diff
@@ -45,6 +47,90 @@ class TestDownloadableIODescriptors(ShotgunTestBase):
     Tests the ability of the descriptor to download to a path on disk.
     """
 
+    # In Python 3.8+, the TestCase class cannot be serialized due to a regression.
+    # It appears an argument parser is reachable from the TestCase instance. The
+    # multiprocessing module serializes both the test method and the instance
+    # associated to it (which makes sense, since we need the test data on the
+    # instance!). This becomes an issue, because ArgumentParser is not picklable
+    # in Python 3.8 (due to a regression it seems). The cleanest way to avoid
+    # this serialization issue without mucking around with the TestCase internals
+    # and that generates the smallest diff is to create a new Implementation
+    # class defined below that reimplements the necessary interface from
```

Comment:
It's difficult for me to know without becoming more familiar with ShotgunTestBase, but are we saying that the only things that needed reimplementing were the assert methods?
I'm not seeing any issues here however, looks reasonably clean.

Reply:
Yeah, that's the only thing that needed to be reimoplemented.